Skarnet update

mobinmob opened the pull request:
<!-- Uncomment relevant sections and delete options which are not applicable -->

#### Testing the changes
- I tested the changes in this PR: **YES**

--
- Update to skarnet packages and manpages.
- A new package for s6-rc-man-pages is added and the man pages repos are now hosted on sourcehut, so homepage and distfiles are changed.

<!--
#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**|**NO**
-->

<!-- Note: If the build is likely to take more than 2 hours, please add ci skip tag as described in
https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#continuous-integration
and test at least one native build and, if supported, at least one cross build.
Ignore this section if this PR is not skipping CI.
-->
<!--
#### Local build testing
- I built this PR locally for my native architecture, (ARCH-LIBC)
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - aarch64-musl
  - armv7l
  - armv6l-musl
-->

mobinmob commented:
- Rebased on master.
- Added the mdev changes @dkwo made in https://github.com/void-linux/void-packages/pull/42488 - many thanks!

Duncaen commented:
I don't like that we start vendoring random configuration files.

Duncaen commented:
I would prefer if you do changes like that in separate PRs from just the bulk updates.

Duncaen commented on srcpkgs/s6-rc-man-pages/template:
```diff
@@ -0,0 +1,16 @@
+maintainer="mobinmob <mobinmob@disroot.org>"
+license="ISC"
+homepage="https://git.sr.ht/~flexibeast/s6-rc-man-pages"
+distfiles="https://git.sr.ht/~flexibeast/s6-rc-man-pages/archive/v0.5.4.1.2.tar.gz"
```
This should use `${version}`.

mobinmob replied:
Done, thanks :)

Duncaen commented on srcpkgs/s6-rc-man-pages/template:
```diff
@@ -0,0 +1,16 @@
+version=0.5.4.1.1
+revision=1
+build_style=gnu-makefile
+hostmakedepends="mdocml"
```
Is the hostmakedepends actually required? The makefiles for this and the other -man-pages packages seem to just install the files.

mobinmob replied:
It is not... 
This is probably something I added during testing or on a previous version that was needed and copied it on new templates:(  Removed, thanks!

mobinmob commented:
> I would prefer if you do changes like that in separate PRs from just the bulk updates.

I agree, but there should be a way to have some configuration files that are tested, even if they are not endorsed in some way for the distribution. This was tested by dkwo. I removed it and corrected the template and service to account for it. Maybe installing the file with vdoc as mdev.conf.sample and making clear in a comment that this is offered as an example will be acceptable?

mobinmob commented:
- Rebased on master.
- Fixes (many thanks for the review @Duncaen !)
- Added `--enable-pedantic-posix` to the execline configure_args.

Duncaen commented on srcpkgs/mdevd/files/mdevd/run:
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+
+exec 2>&1
+exec mdevd -b 200000 -O4
```
```shell
#!/bin/sh

exec 2>&1
[ -r ./conf ] && . ./conf
exec mdevd ${OPTS:- -b 200000 -O4}
```

mobinmob replied:
Done, many thanks!

mobinmob commented:
> * Added `--enable-pedantic-posix` to the execline configure_args.

That is needed, because in certain cases the execline cd is called instead of the shell internal command and that works in different way...

dkwo commented:
I tested that config for mdev a while ago, but dropping it or including as vdoc seems also fine.
Note that withot a config it is rather useless.

mobinmob commented:
> I tested that config for mdev a while ago, but dropping it or including as vdoc seems also fine.
> Note that without a config it is rather useless.

I think that mdevd has a pretty clear log message for a missing configuration file, so I hope a sample config under the usual place for documentation will be helpful.

mobinmob commented:
bump...

mobinmob commented:
Superseeded by https://github.com/void-linux/void-packages/pull/47177 .
